pkg/ebpf/dns_stats.go

```diff
@@ -34,36 +38,89 @@ type dnsPacketInfo struct {
 	type_         DNSPacketType
 }
 
+type stateKey struct {
+	key dnsKey
+	id  uint16
+}
+
 type dnsStatKeeper struct {
-	mux   sync.Mutex
-	stats map[dnsKey]dnsStats
+	mux              sync.Mutex
+	stats            map[dnsKey]dnsStats
+	state            map[stateKey]time.Time
+	expirationPeriod time.Duration
+	exit             chan struct{}
+	maxSize          int // maximum size of the state map
+	deleteCount      int
+	deleteThreshold  int
 }
 
-func newDNSStatkeeper() *dnsStatKeeper {
-	return &dnsStatKeeper{
-		stats: make(map[dnsKey]dnsStats),
+func newDNSStatkeeper(timeout time.Duration) *dnsStatKeeper {
+	statsKeeper := &dnsStatKeeper{
+		stats:            make(map[dnsKey]dnsStats),
+		state:            make(map[stateKey]time.Time),
+		expirationPeriod: timeout,
+		exit:             make(chan struct{}),
+		maxSize:          10000,
```

Comment:
It would be good to know either with a comment or benchmark how much memory this would take up.